test-requirements.txt

```diff
@@ -1,9 +1,7 @@
-mock
-nose
 coverage
-pep8
-pylint >= 1.1.0
 ddt
 diff-cover >= 0.2.1
--e .
--r requirements.txt
+edx-lint>=0.4.0,<1.0.0
+mock
```

Comment:
Done. @benpatterson thoughts on using ranges rather than exact versions for test requirements?